src/Microsoft.IdentityModel.Clients.ActiveDirectory/Core/Cache/Items/MsalAccessTokenCacheItem.cs

```diff
@@ -1,131 +1,138 @@
+            return FromJObject(JObject.Parse(json));
+        }
+
+        internal static MsalAccessTokenCacheItem FromJObject(JObject j)
         {
-            get
+            long cachedAt = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.CachedAt);
+            long expiresOn = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExpiresOn);
+
+            // This handles a bug with the name in previous MSAL.  It used "ext_expires_on" instead of
+            // "extended_expires_on" per spec, so this works around that.
+            long ext_expires_on = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExtendedExpiresOn_MsalCompat);
+            long extendedExpiresOn = JsonUtils.ExtractParsedIntOrZero(j, StorageJsonKeys.ExtendedExpiresOn);
+            if (extendedExpiresOn == 0 && ext_expires_on > 0)
             {
-                DateTime dtDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
-                var expiresLong = Convert.ToInt64(ExpiresOnUnixTimestamp, CultureInfo.InvariantCulture);
-                return dtDateTime.AddSeconds(expiresLong).ToUniversalTime();
+                extendedExpiresOn = ext_expires_on;
             }
+
+            var item = new MsalAccessTokenCacheItem
+            {
+                TenantId = JsonUtils.ExtractExistingOrEmptyString(j, StorageJsonKeys.Realm),
+                NormalizedScopes = JsonUtils.ExtractExistingOrEmptyString(j, StorageJsonKeys.Target),
+                CachedAt = cachedAt.ToString(CultureInfo.InvariantCulture),
+                ExpiresOnUnixTimestamp = expiresOn.ToString(CultureInfo.InvariantCulture),
+                ExtendedExpiresOnUnixTimestamp = extendedExpiresOn.ToString(CultureInfo.InvariantCulture),
+                UserAssertionHash = JsonUtils.ExtractExistingOrEmptyString(j, StorageJsonKeys.UserAssertionHash),
+            };
+
+            item.PopulateFieldsFromJObject(j);
+
+            return item;
         }
 
-        internal DateTimeOffset ExtendedExpiresOn
+        internal override JObject ToJObject()
         {
-            get
-            {
-                DateTime dtDateTime = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
-                var expiresLong = Convert.ToInt64(ExtendedExpiresOnUnixTimestamp, CultureInfo.InvariantCulture);
+            var json = base.ToJObject();
 
-                return dtDateTime.AddSeconds(expiresLong).ToUniversalTime();
-            }
+            json[StorageJsonKeys.Realm] = TenantId;
+            json[StorageJsonKeys.Target] = NormalizedScopes;
+            json[StorageJsonKeys.UserAssertionHash] = UserAssertionHash;
+            json[StorageJsonKeys.CachedAt] = CachedAt;
+            json[StorageJsonKeys.ExpiresOn] = ExpiresOnUnixTimestamp;
+            json[StorageJsonKeys.ExtendedExpiresOn] = ExtendedExpiresOnUnixTimestamp;
+
+            // previous versions of msal used "ext_expires_on" instead of the correct "extended_expires_on".
+            // this is here for back compat
```

Comment:
see comment above..maybe even define the versions of msal using "ext_expires_on" because as we move ahead, this might get confusing and shortening it as well.